### PR TITLE
Add ``redundant-u-string-prefix`` checker

### LINKED ISSUE #4102: Add check for redundant u prefixed string in Python 3
### Is your feature request related to a problem? Please describe

In Python 3, the `u` prefix is no longer necessary to indicate that a string is unicode, but some programmers may put it there out of habit or because they copied the code from somewhere else.

### Describe the solution you'd like

Pylint should flag instances of u-prefixed strings in Python 3 with the `redundant-string-prefix` code.

---

I can contribute this check if there is interest in it.

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

This adds a checker for u-prefixes for strings, as used in Python 2.

Closes #4102

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -89,10 +89,14 @@ Release date: TBA
 
   Closes #626
 
-* Add ``disable-next`` option: allows using `# pylint: disable-next=msgid` to disable a message for the following line
+* Added ``disable-next`` option: allows using `# pylint: disable-next=msgid` to disable a message for the following line
```
Comment (DanielNoord) [inline ChangeLog:92]: All other changes also use past tense

**Review by Pierre-Sassoulas — APPROVED** [review]
:+1: Neat clean feature, thank you !

**Comment by DanielNoord** [comment]
Could a maintainer force re-run the checks? Something went wrong with the runners I think..

**Comment by Pierre-Sassoulas** [comment]
I relaunched, CodeQL failed again but it's not a mandatory check. The pypy one seems genuine.

**Comment by DanielNoord** [comment]
Oops, looks like I forgot to run the test suite locally. I think I didn't expect pylint to use u-prefixes.. 😅

**Comment by DanielNoord** [comment]
Turns out `astroid.Const.kind` is only supported Python >=3.8.
@Pierre-Sassoulas do you know if there is a quick and easy way to infer the `string kind` on <=3.7?

**Comment by Pierre-Sassoulas** [comment]
I don't know if it would be "easy" or "elegant", but it would be a change in astroid. Maybe we can do this check only in python 3.8+ ?

**Comment by DanielNoord** [comment]
Sure!

**Review by Pierre-Sassoulas — APPROVED** [review]
Great new checker !

**Pull request merged**